module/User/src/Authorization/GenericAclService.php

```diff
@@ -33,17 +34,27 @@ public function __construct(
 
     /**
      * @inheritDoc
+     *
+     * The role that should take precedence should be returned first.
+     * This is because of the behaviour of {@link \Laminas\Permissions\Acl\Role\Registry}.
      */
     protected function getRole()
     {
         if ($this->authService->hasIdentity()) {
-            return $this->authService->getIdentity();
+            $user = $this->authService->getIdentity();
+            if ($user instanceof User) {
```

Comment:
I understand the additional check (`instanceof`), however, if the `authService` has an identity it is guaranteed to be a `User`.

Also, I think it is a bit nicer to have an empty line between the two lines.